src/App.tsx

```diff
@@ -139,7 +139,7 @@ export default class App extends Component<{}, { chosenNode: NodeID, value: stri
                 {// TODO: get rid of private access
                 }
                 <label>Selected node: {this.state.ourTree._selectNodeById(this.state.chosenNode).data.name} : {this.state.chosenNode}</label>
-                <button onClick = { e => this.remove(e) }>Usuń</button>
+                <button onClick = { e => this.remove(e) } disabled={this.state.ourTree.isRootId(this.state.chosenNode)}>Usuń</button>
```

Comment:
Does this actually prevent dispatching the event?

Reply:
Yes it does.